### ISSUE TITLE
conventional-commits bumps 0.x versions incorrectly

### ISSUE THREAD

**Issue by aleclarson** [intuit/auto#1699]
**Describe the bug**

When the version is `0.1.0` and a `feat:` commit is merged, the version is bumped to `0.2.0` instead of `0.1.1`.

This is wrong, as the major version (as far as NPM interprets it) is the leftmost, non-zero digit.

Essentially, `feat:` and `fix:` commits are identical for `0.x` versions.

**Environment information:**

```txt
"auto" version: v10.5.0
"git"  version: v2.19.2
"node" version: v14.15.3
```

**Comment by hipstersmoothie** [comment]
Happy to do this. having trouble finding this in the spec though

**Comment by hipstersmoothie** [comment]
Okay I see what you're saying now. `^0.2.0` doesn't include `0.3.0` or anything like that. only changes to the rightmost digit.

![Screen Shot 2021-01-05 at 3 24 28 PM](https://user-images.githubusercontent.com/1192452/103710545-1d33e500-4f6a-11eb-91d9-77f8d8222322.png)

Question: If both `minor` and `patch` both increment the rightmost digit how do you go from `0.0.0` => `0.1.0`? Both a  `minor` and `patch` would result in `0.0.1`.  `major` would result in `1.0.0` so you could never increment the middle digit

**Comment by hipstersmoothie** [comment]
There is a lot of reading to do around this: https://github.com/semver/semver/issues/221

This would be a pretty big change to how the semantic versioning works right now. What I find odd is that npm treats semver in two ways:

1. in caret ranges it does what you (and the posted issue) describe
2. when using `npm version` it doesn't respect any of these conventions and just bumps the relevant digit without regard to 0.x or 0.y.x

**Comment by hipstersmoothie** [comment]
> Another solution might be to create a plugin that aliases/downgrades commit bumps

This seems to work okay, what are your thoughts @aleclarson?

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v10.13.0`](https://github.com/intuit/auto/releases/tag/v10.13.0) :rocket:
